### PR TITLE
Clean up and update park.json

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Infrastructure "Clean up and modernize parks.json"```

#### Purpose of change
I wanted to add an overrun zombie killzone in a park, but the file was a mess so I had to fix that first.

#### Describe the solution
Removes outdated terms like f_null
Changes a lot of place_items and place_monsters to by letter designation
Reorganizes old ported maps that were sorted in alphabetical order instead of human-readable.
Adds variable terrain, eg. making random flower choices instead of placing dandelions and things all in the same place

#### Describe alternatives you've considered
I don't really have to do this, the maps work fine, but I can't not.

#### Testing
Tested, works fine. Parks aren't exactly like they were before but they're not too changed.

#### Additional context
Depending on how big this gets, I still might add an overrun fortification as part of this.

### CODE REVIEW

File: data/json/mapgen/park.json [hunk 1]
```diff
@@ -336,198 +248,160 @@
...
+      ],
+      "palettes": [ "park_scenic_palette" ],
       "place_items": [
         { "chance": 8, "item": "dogfight", "x": 2, "y": 12 },
         { "chance": 30, "item": "trash", "x": 6, "y": 10 },
         { "chance": 10, "item": "child_items", "x": 7, "y": 18 },
         { "chance": 3, "item": "creepy", "x": 22, "y": 21 },
         { "chance": 10, "item": "trash_forest", "x": 10, "y": 23 },
-        { "chance": 20, "item": "trash_forest", "x": 18, "y": 22 },
+        { "chance": 500, "item": "trash_forest", "x": 18, "y": 22 },
```
Comment (BevapDin) [inline data/json/mapgen/park.json]: Values that large are not valid here. Valid values are in the range 1 to 100.

Comment (AMurkin) [reply data/json/mapgen/park.json]: Reported in #36637

Comment (ZhilkinSerg) [reply data/json/mapgen/park.json]: @I-am-Erk Should it be `50` (like in 50%) opposed to `500` (1 in 500)?

Comment (I-am-Erk) [reply data/json/mapgen/park.json]: Yes, that looks like an extra zero typo

**Pull request merged**